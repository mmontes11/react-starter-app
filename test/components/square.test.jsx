import React from "react";
import { shallow } from "enzyme";
import Square from "components/square";

describe("components/square", () => {
  const onClick = jest.fn();
  const wrapper = shallow(<Square value="X" onClick={() => onClick()} />);

  it("renders a square", () => {
    expect(wrapper.text()).toBe("X");
    expect(wrapper).toMatchSnapshot();
  });
  it("simulares clicks", () => {
    wrapper.find("button").simulate("click");

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
