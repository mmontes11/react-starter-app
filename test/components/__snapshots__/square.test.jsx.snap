// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/square renders a square 1`] = `
<button
  className="square"
  onClick={[Function]}
>
  X
</button>
`;
